No way to configure blas support

Whenever I try to configure pythran to use within a package setting, it just cannot seem to figure out where BLAS is. 

> ... 
> dot.hpp:18:10: fatal error: 'cblas.h' file not found

I have a simple .py file than compiles fine with pythran via: 
```bash
PYTHRANRC=.pythranrc pythran src/myproject/_lanczos_pythran.py 
```
Where my .pythranrc is: 

```
[compiler]
cflags=-std=c++17
blas=openblas
include_dirs=/usr/local/opt/openblas/include
library_dirs=/usr/local/opt/openblas/lib
```

It also works fine if I use:
```bash
pythran --config pythran-openblas src/myproject/_lanczos_pythran.py 
```

But it just doesnt work in the package build setting; nothing propagates. 

> python build . --no-isolation --wheel

Things I've tried: 

1. Generator + using a meson environment
```meson 
pythran = find_program('pythran', native: true, version: '>=0.14.0')
pythran_env = environment()
pythran_env.set('PYTHRANRC', join_paths(meson.project_source_root(), '.pythranrc'))
pythran_gen = generator(
    pythran,
    output : '@BASENAME@.cpp',
    arguments : ['-E', '@INPUT@', '-o', '@OUTPUT@'],
)
pythran_gen.process('_lanczos_pythran.py', env: pythran_env)
```

2. Custom target + command + using a meson environment

```meson 
cmd = [pythran, '-E', '@INPUT@', '-o', '@OUTPUT@']
_lanczos_pythran_cpp = custom_target(
  '_lanczos_pythran_cpp',
  output: '_lanczos_pythran.cpp',
  input: '_lanczos_pythran.py',
  command: cmd,
  env: pythran_env
)
```

3. Manually set command + pythran-openblas 

```meson 
cmd = [pythran, '--config', 'compiler.blas=pythran-openblas', '-E', '@INPUT@', '-o', '@OUTPUT@']
_lanczos_pythran_cpp = custom_target(
  '_lanczos_pythran_cpp',
  output: '_lanczos_pythran.cpp',
  input: '_lanczos_pythran.py',
  command: cmd
)
```

--- a/.github/workflows/core.yml
+++ b/.github/workflows/core.yml
@@ -26,6 +26,7 @@ jobs:
         pip install scipy-openblas64
         pip install pytest-xdist
         sudo apt install ${{ matrix.cpp-version }}
+        sudo apt install libopenblas-dev  # for meson integration testing
         if test ${{ matrix.python-version }} != 'pypy-3.9'; then pip install scipy ; fi
     - name: Setup
       run: |

--- a/docs/MANUAL.rst
+++ b/docs/MANUAL.rst
@@ -402,6 +402,15 @@ you can change its base class by using ``PythranBuildExt[base_cls]`` instead.
 
     This behavior is likely to change with revisions >= 1.
 
+Meson Integration
+*****************
+
+Pythran provides helper programs to easien integration with meson, through
+``pythran-config``. The following project is tested during validation and seems
+to work as expected. It includes support for a custom ``pythranrc`` file:
+
+.. literalinclude:: ../pythran/tests/test_distutils/meson.build
+
 
 Cross compilation
 *****************

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -27,7 +27,7 @@ Changelog = "https://pythran.readthedocs.io/en/latest/Changelog.html"
 
 [project.optional-dependencies]
 doc = ["numpy", "nbsphinx", "scipy", "guzzle_sphinx_theme"]
-test = ["ipython", "nbval", "cython", "wheel", "packaging"]
+test = ["ipython", "nbval", "cython", "wheel", "packaging", "ninja", "meson"]
 
 [project.scripts]
 pythran = "pythran.run:run"

--- a/pythran/config.py
+++ b/pythran/config.py
@@ -398,14 +398,16 @@ def run():
                        for include in extension['libraries'])
 
         if args.python:
-            libpl = distutils.sysconfig.get_config_var('LIBPL')
-            if libpl:
-                ldflags.append('-L' + libpl)
             libs = distutils.sysconfig.get_config_var('LIBS')
             if libs:
                 ldflags.extend(shsplit(libs))
-            ldflags.append(compiler_obj.library_option('python')
-                           + distutils.sysconfig.get_config_var('VERSION'))
+            libdir = distutils.sysconfig.get_config_var('LIBDIR')
+            pylib = distutils.sysconfig.get_config_var('LDLIBRARY')
+            if libdir and pylib:
+                ldflags.append(os.path.join(libdir, pylib))
+            else:
+                ldflags.append(compiler_obj.library_option('python')
+                               + distutils.sysconfig.get_config_var('VERSION'))
 
         logger.info('LDFLAGS = '.rjust(10) + ' '.join(ldflags))
         if args.libs:

--- a/pythran/config.py
+++ b/pythran/config.py
@@ -272,18 +272,8 @@ def make_extension(python, **extra):
         extension['define_macros'].append('PYTHRAN_BLAS_NONE')
 
     if user_blas not in reserved_blas_entries:
-        try:
-            import numpy.distutils.system_info as numpy_sys
-             # Numpy can pollute stdout with checks
-            with silent():
-                numpy_blas = numpy_sys.get_info(user_blas)
-                extension['libraries'].extend(numpy_blas.get('libraries', []))
-                extension['library_dirs'].extend(
-                    numpy_blas.get('library_dirs', []))
-        # `numpy.distutils` not present for Python >= 3.12
-        except ImportError:
-            extension["libraries"].append(user_blas)
-            extension['define_macros'].append('PYTHRAN_BLAS_{}'.format(user_blas.upper()))
+        extension["libraries"].append(user_blas)
+        extension['define_macros'].append('PYTHRAN_BLAS_{}'.format(user_blas.upper()))
 
 
     # final macro normalization

--- a/pythran/tests/test_distutils.py
+++ b/pythran/tests/test_distutils.py
@@ -170,3 +170,31 @@ class TestDistutils(unittest.TestCase):
         demo_so = find_so(r"a.*\.so", dist_path)
         self.assertIsNotNone(demo_so)
         shutil.rmtree(dist_path)
+
+try:
+    check_call(['ninja', '--version'])
+    has_ninja = True
+except:
+    has_ninja = False
+
+try:
+    check_call(['meson', '--version'])
+    has_meson = True
+except:
+    has_meson = False
+
+class TestMeson(unittest.TestCase):
+
+    @unittest.skipIf(not has_meson, "meson not found")
+    @unittest.skipIf(not has_ninja, "ninja not found")
+    def test_meson_build(self):
+        srcdir = os.path.join(cwd, 'test_distutils')
+        builddir = os.path.join(srcdir, '_meson_build')
+        shutil.rmtree(builddir, ignore_errors=True)
+        check_call(['meson', 'setup', builddir, '.'],
+                   cwd=srcdir)
+        check_call(['ninja', '-v'],
+                   cwd=builddir)
+        check_call([python, '-c', 'import b'],
+                   cwd=builddir)
+

--- a/pythran/tests/test_distutils/meson.build
+++ b/pythran/tests/test_distutils/meson.build
@@ -1,0 +1,31 @@
+project('pythran-test', 'cpp')
+
+pymod = import('python')
+py = pymod.find_installation(modules:['pythran'])
+
+pythran_cmd = [py.path(), '-m', 'pythran.run', '-E', '@INPUT@', '-o', '@OUTPUT@']
+
+pythranrc = join_paths(meson.project_source_root(), 'pythran.rc')
+
+pythran_env = environment()
+pythran_env.set('PYTHRANRC', pythranrc)
+
+b_cpp = custom_target(
+  'b_cpp',
+  output : 'b.cpp',
+  input: 'b.py',
+  command: pythran_cmd,
+  env: pythran_env,
+  depend_files: [pythranrc],
+)
+
+pythran_cppflags = run_command(py.path(), '-m', 'pythran.config', '--cflags', check:true, env:pythran_env).stdout().strip().split()
+pythran_ldflags = run_command(py.path(), '-m', 'pythran.config', '--libs', check:true, env:pythran_env).stdout().strip().split()
+
+py.extension_module(
+  'b',
+  b_cpp,
+  cpp_args: pythran_cppflags,
+  link_args: pythran_ldflags,
+  install: true,
+)

--- a/pythran/tests/test_distutils/pythran.rc
+++ b/pythran/tests/test_distutils/pythran.rc
@@ -1,0 +1,3 @@
+[compiler]
+blas=openblas
+include_dirs=/usr/include/openblas

